src/J2N/compatibility/IReadOnlyDictionary.cs

```diff
@@ -0,0 +1,26 @@
+﻿#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
```

Comment:
Do not tell me how to solve the situation with `ReadOnlyDictionaryTests.cs`, `IReadOnlyDictionary` is used inside?

Reply:
I see your point.

[`ReadOnlyDictionary<TKey, TValue>` did not exist in .NET Framework 4.0](https://docs.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.readonlydictionary-2?view=netframework-4.8). However, it is used by ICU4N through the `DictionaryExtensions.AsReadOnly()` method, so we cannot simply exclude that class. That said, ICU4N does not cast to the `IReadOnlyDictionary<TKey, TValue>` interface, it uses `IDictionary<TKey, TValue>` and `ICollection<T>` only. So, we can exclude the interface from J2N on .NET Framework 4.0 with no issue.

There are 2 different approaches that can be used here and I will accept either:

**Approach 1:**

Leave the `IReadOnlyDictionary<TKey, TValue>` interface, but mark it **internal**.

Compilation features are **enabled** based on what APIs are available in the underlying target framework, so we would need a `FEATURE_IREADONLYDICTIONARY_GENERIC` constant that exists for all frameworks *except* .NET Framework 4.0 and a `#if !FEATURE_IREADONLYDICTIONARY_GENERIC` block to include `IReadOnlyDictionary<TKey, TValue>` in the compile when the feature is not enabled.

Also, please move `IReadOnlyDictionary<TKey, TValue>` back to the `System.Collections.Generic` namespace so we don't need so many conditional using directives throughout the project.

**Approach 2:**

Conditionally replace `IReadOnlyDictionary<TKey, TValue>` with `IDictionary<TKey, TValue>` in the tests.

It looks like the only things that would need to conditionally compile are the `collection` member variable, the constructor, `VerifyEnumerator`, `VerifyGenericEnumerator` and `VerifyCollection` methods of the [IReadOnlyDictionary_T_Test<TKey, TValue>](https://github.com/NightOwl888/J2N/blob/master/tests/J2N.Tests.xUnit/Collections/ObjectModel/ReadOnlyDictionaryTests.cs#L439) class. Since `IDictionary<TKey, TValue>` is a superset of `IReadOnlyDictionary<TKey, Value>`, I believe it could be used as a replacement for the test fixture in those methods. The `ReadOnlyDictionary<TKey, TValue>` class implements `IDictionary<TKey, TValue>`, so that shouldn't be an issue. The rest of the methods in `ReadOnlyDictionaryTests.cs` don't seem to utilize the `IReadOnlyDictionary<TKey, Value>` interface.

Reply:
Thank you very much for your help.
I chose the second approach.

Reply:
I will fix this in the next commit.